File: samples/snippets/visualbasic/VS_Snippets_Wpf/brushsamples_procedural_snip/VisualBasic/MyApp.vb
Imports System
Imports System.Windows
Imports System.Windows.Navigation
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports System.Windows.Controls

Namespace Microsoft.Samples.BrushExamples

	' Displays the sample.
	Public Class MyApp
		Inherits Application

		Public Sub New()
			 AddHandler AppDomain.CurrentDomain.UnhandledException, AddressOf currentDomain_UnhandledException
		End Sub


		Protected Overrides Sub OnStartup(ByVal e As StartupEventArgs)
			MyBase.OnStartup(e)
			CreateAndShowMainWindow()
		End Sub


		Private Sub CreateAndShowMainWindow()
			' Create the application's main window.
			Dim myWindow As New NavigationWindow()

			' Display the sample
			Dim myContent As Page = New SampleViewer()
			myWindow.Navigate(myContent)
			MainWindow = myWindow
			myWindow.Show()
		End Sub

		Private Sub currentDomain_UnhandledException(ByVal sender As Object, ByVal args As UnhandledExceptionEventArgs)
			MessageBox.Show("Unhandled exception: " & args.ExceptionObject.ToString())
		End Sub
	End Class

	' Starts the application.
	Friend NotInheritable Class EntryClass
		<System.STAThread()>
		Private Shared Sub Main()

			Dim app As New MyApp()
			app.Run ()
		End Sub
	End Class
End Namespace

File: samples/snippets/csharp/VS_Snippets_Wpf/brushsamples_procedural_snip/CSharp/MyApp.cs
using System;
using System.Windows;
using System.Windows.Navigation;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Windows.Controls;

namespace Microsoft.Samples.BrushExamples
{

    // Displays the sample.
    public partial class MyApp : Application
    {

        public MyApp()
        {
             AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(currentDomain_UnhandledException);
        }

        protected override void OnStartup (StartupEventArgs e)
        {
            base.OnStartup(e);
            CreateAndShowMainWindow();
        }

        private void CreateAndShowMainWindow ()
        {
            // Create the application's main window.
            NavigationWindow myWindow = new NavigationWindow();

            // Display the sample
            Page myContent = new SampleViewer();
            myWindow.Navigate(myContent);
            MainWindow = myWindow;
            myWindow.Show();
        }

        private void currentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs args)
        {
            MessageBox.Show("Unhandled exception: " + args.ExceptionObject.ToString());
        }
    }

    // Starts the application.
    internal sealed class EntryClass
    {
        [System.STAThread()]
        private static void Main ()
        {

            MyApp app = new MyApp ();
            app.Run ();
        }
    }
}
